src/evalcontext.cc

```diff
@@ -35,19 +35,36 @@ ValuePtr EvalContext::getArgValue(size_t i, const Context *ctx) const
   Resolves arguments specified by evalctx, using args to lookup positional arguments.
   Returns an AssignmentMap (string -> Expression*)
 */
-AssignmentMap EvalContext::resolveArguments(const AssignmentList &args) const
+AssignmentMap EvalContext::resolveArguments(const AssignmentList &args, const AssignmentList &optargs, bool usermodule) const
 {
   AssignmentMap resolvedArgs;
   size_t posarg = 0;
+  bool tooManyWarned=false;
   // Iterate over positional args
   for (size_t i=0; i<this->numArgs(); i++) {
     const auto &name = this->getArgName(i); // name is optional
     const auto expr = this->getArgs()[i].expr.get();
     if (!name.empty()) {
+      if(name.at(0)!='$' && (OpenSCAD::parameterCheck || (not usermodule))){
```

Comment:
It feels to me that the `OpenSCAD::parameterCheck` and `usermodule` check should be external to this function and it should rather be passed an single bool, e.g. `silent`. Does that make sense to you?
I believe we have another `silent` flag like that somewhere else.

Reply:
we have a `silent` flag in the variable lookup.

Adding a silent flag to resolveArguments is fine by me, must of the time it is called indirectly by setVariables.

Reply:
The idea would be to move any global state checks as far up in the call hierarchy as we can, so that lower-level functions are less dependent on global state and thus easier to test or refactor later.